src/Compilers/VisualBasic/Portable/Symbols/TypeSymbolExtensions.vb

```diff
@@ -291,7 +291,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
 
                 If Not (t1IsDefinition AndAlso t2IsDefinition) Then ' This is a generic instantiation case
 
-                    If t1.OriginalDefinition <> t2.OriginalDefinition Then
+                    If Not TypeSymbol.Equals(t1.OriginalDefinition, t2.OriginalDefinition, TypeCompareKind.ConsiderEverything) Then
```

Comment:
>Not TypeSymbol.Equals(t1.OriginalDefinition, t2.OriginalDefinition, TypeCompareKind.ConsiderEverything) [](start = 23, length = 103)

```!t1.OriginalDefinition.Equals(t2.OriginalDefinition)```?

I think it should be fine to use this overload of Equals, the primary goal is to avoid using ```==```/```<>``` operators.